proxy.c

```diff
@@ -206,7 +215,7 @@ void handle_client(int client_sock, struct sockaddr_in client_addr)
 {
     if ((remote_sock = create_connection()) < 0) {
         perror("Cannot connect to host");
-        return;
+        goto out;
```

Comment:
For error handling in C, since it lacks try/catch blocks, I personally think using a cleanup goto is ideal. Maybe the label name is wrong, it could be named err: or cleanup:?

If you insist I'll add 2 more close() lines before return;.